app/assets/src/components/utils/sample.js

```diff
@@ -1,21 +1,38 @@
 import { last } from "lodash/fp";
 
-export const pipelineVersionHasAssembly = pipelineVersion => {
+export const pipelineVersionAtLeast = (pipelineVersion, testVersion) => {
   if (!pipelineVersion) return false;
-  const versionNums = pipelineVersion.split(".");
-  return (
-    +versionNums[0] >= 4 || (+versionNums[0] === 3 && +versionNums[1] >= 1)
-  );
-};
 
-export const pipelineVersionHasCoverageViz = pipelineVersion => {
-  if (!pipelineVersion) return false;
-  const versionNums = pipelineVersion.split(".");
-  return (
-    +versionNums[0] >= 4 || (+versionNums[0] === 3 && +versionNums[1] >= 6)
-  );
+  // turn undefined to 0.
+  const toInt = versionNumber => +versionNumber || 0;
+
+  const pipelineNums = pipelineVersion.split(".");
+  const testNums = testVersion.split(".");
+
+  if (toInt(pipelineNums[0]) > toInt(testNums[0])) {
+    return true;
+  } else if (toInt(pipelineNums[0]) === toInt(testNums[0])) {
+    if (toInt(pipelineNums[1]) > toInt(testNums[1])) {
+      return true;
+    } else if (toInt(pipelineNums[1]) === toInt(testNums[1])) {
+      if (toInt(pipelineNums[2]) >= toInt(testNums[2])) {
```

Comment:
for pipelineNums[2] I might've seen some version strings passed along without the third 'digit'.. something to look out for